pkg/utils/db.go

```diff
@@ -135,8 +136,38 @@ func GetSlaveServerID(ctx context.Context, db *sql.DB) (map[uint32]struct{}, err
 	return serverIDs, nil
 }
 
+var (
+	useMasterStatusCache bool
+	masterMu             sync.Mutex
+	masterStatusCached   bool
+	masterPosCache       gmysql.Position
+	masterGTIDCache      gtid.Set
+)
+
+// EnableMasterStatusCache starts caching result and use it of GetMasterStatus
+func EnableMasterStatusCache() {
+	masterMu.Lock()
+	defer masterMu.Unlock()
+	useMasterStatusCache = true
+	masterStatusCached = false
+}
+
+// DisableMasterStatusCache stops caching result and use it of GetMasterStatus
+func DisableMasterStatusCache() {
+	masterMu.Lock()
+	defer masterMu.Unlock()
+	useMasterStatusCache = false
+}
+
 // GetMasterStatus gets status from master
 func GetMasterStatus(db *sql.DB, flavor string) (gmysql.Position, gtid.Set, error) {
+	masterMu.Lock()
+	defer masterMu.Unlock()
+
+	if useMasterStatusCache && masterStatusCached {
+		return masterPosCache, masterGTIDCache, nil
```

Comment:
Will this cause `global point` [greater](https://github.com/pingcap/dm/blob/7ec79aed1816481099f01fcf0aaf48c7b783c96a/syncer/status.go#L79) than `master point`?